// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api.autowalk;
option go_package = "bosdyn/api/autowalk/autowalk_service";

option java_outer_classname = "AutowalkServiceProto";

import "bosdyn/api/data_chunk.proto";

service AutowalkService {
    // Compile a walk into a mission.
    // Input DataChunks should deserialize into a CompileAutowalkRequest.
    // Output DataChunks should deserialize into a CompileAutowalkResponse.
    // This rpc is stateless.
    rpc CompileAutowalk(stream DataChunk) returns (stream DataChunk) {}

    // Compile a walk into a mission then load to mission service.
    // Input DataChunks should deserialize into a LoadAutowalkRequest.
    // Output DataChunks should deserialize into a LoadAutowalkResponse.
    rpc LoadAutowalk(stream DataChunk) returns (stream DataChunk) {}
}